File: src/com/koxudaxi/pydantic/PydanticConfigPanel.java
package com.koxudaxi.pydantic;

import com.intellij.ide.BrowserUtil;
import com.intellij.ide.plugins.newui.EmptyCaret;
import com.intellij.openapi.project.Project;
import com.intellij.ui.ColorUtil;
import com.intellij.util.ui.JBUI;
import com.intellij.util.ui.UIUtil;

import javax.swing.*;
import javax.swing.event.HyperlinkEvent;
import javax.swing.text.EditorKit;
import javax.swing.text.html.HTMLEditorKit;
import javax.swing.text.html.StyleSheet;


public class PydanticConfigPanel {

    PydanticConfigPanel(Project project) {
        PydanticConfigService pydanticConfigService = PydanticConfigService.Companion.getInstance(project);

        this.initTypedCheckBox.setSelected(pydanticConfigService.getInitTyped());
        this.warnUntypedFieldsCheckBox.setSelected(pydanticConfigService.getWarnUntypedFields());

        boolean enableInitTypedCheckBox = pydanticConfigService.getMypyInitTyped() == null;
        this.initTypedCheckBox.setEnabled(enableInitTypedCheckBox);
        this.ifEnabledIncludeTheTextPane.setEnabled(enableInitTypedCheckBox);

        boolean warnUntypedFieldsCheckBox = pydanticConfigService.getMypyWarnUntypedFields() == null;
        this.warnUntypedFieldsCheckBox.setEnabled(warnUntypedFieldsCheckBox);
        this.ifEnabledRaiseATextPane.setEnabled(warnUntypedFieldsCheckBox);

        setHyperlinkHtml(this.textPane1, "See <a href=\"https://koxudaxi.github.io/pydantic-pycharm-plugin/\">documentation</a> for more details.</p>");
    }

    private JPanel configPanel;
    private JCheckBox initTypedCheckBox;
    private JTextPane ifEnabledIncludeTheTextPane;
    private JCheckBox warnUntypedFieldsCheckBox;
    private JTextPane ifEnabledRaiseATextPane;
    private JTextPane textPane1;

    public Boolean getInitTyped() {
        return initTypedCheckBox.isSelected();
    }

    public Boolean getWarnUntypedFields() {
        return warnUntypedFieldsCheckBox.isSelected();
    }

    public JPanel getConfigPanel() {
        return configPanel;
    }

    private void setHyperlinkHtml(JTextPane jTextPane, String html) {

        jTextPane.setContentType("text/html");
        jTextPane.setEditable(false);
        jTextPane.setEditorKit(UIUtil.getHTMLEditorKit());

        EditorKit kit = jTextPane.getEditorKit();
        if (kit instanceof HTMLEditorKit) {
            StyleSheet css = ((HTMLEditorKit) kit).getStyleSheet();

            css.addRule("a, a:link {color:#" + ColorUtil.toHex(JBUI.CurrentTheme.Link.Foreground.ENABLED) + ";}");
            css.addRule("a:visited {color:#" + ColorUtil.toHex(JBUI.CurrentTheme.Link.Foreground.VISITED) + ";}");
            css.addRule("a:hover {color:#" + ColorUtil.toHex(JBUI.CurrentTheme.Link.Foreground.HOVERED) + ";}");
            css.addRule("a:active {color:#" + ColorUtil.toHex(JBUI.CurrentTheme.Link.Foreground.PRESSED) + ";}");
            css.addRule("body {background-color:#" + ColorUtil.toHex(JBUI.CurrentTheme.DefaultTabs.background()) + ";}");
        }

        jTextPane.setText(html);
        jTextPane.setCaret(EmptyCaret.INSTANCE);
        jTextPane.setBackground(UIUtil.TRANSPARENT_COLOR);
        jTextPane.setOpaque(false);
        jTextPane.addHyperlinkListener(e -> {
            if (e.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
                BrowserUtil.browse(e.getURL());
            }
        });
    }
}

File: src/com/koxudaxi/pydantic/PydanticConfigPanel.kt
package com.koxudaxi.pydantic

import com.koxudaxi.pydantic.PydanticConfigService.Companion.getInstance
import javax.swing.JPanel
import javax.swing.JCheckBox
import javax.swing.JTextPane
import javax.swing.text.html.HTMLEditorKit
import com.intellij.util.ui.JBUI
import com.intellij.ide.plugins.newui.EmptyCaret
import javax.swing.event.HyperlinkEvent
import com.intellij.ide.BrowserUtil
import com.intellij.openapi.project.Project
import com.intellij.ui.ColorUtil
import com.intellij.util.ui.HTMLEditorKitBuilder
import com.intellij.util.ui.UIUtil

class PydanticConfigPanel(project: Project) {
    lateinit var configPanel: JPanel
    private lateinit var initTypedCheckBox: JCheckBox
    private lateinit var ifEnabledIncludeTheTextPane: JTextPane
    private lateinit var warnUntypedFieldsCheckBox: JCheckBox
    private lateinit var ifEnabledRaiseATextPane: JTextPane
    private lateinit var textPane1: JTextPane

    init {
        val pydanticConfigService = getInstance(project)
        initTypedCheckBox.isSelected = pydanticConfigService.initTyped
        warnUntypedFieldsCheckBox.isSelected = pydanticConfigService.warnUntypedFields
        val enableInitTypedCheckBox = pydanticConfigService.mypyInitTyped == null
        initTypedCheckBox.isEnabled = enableInitTypedCheckBox
        ifEnabledIncludeTheTextPane.isEnabled = enableInitTypedCheckBox
        val warnUntypedFieldsCheckBox = pydanticConfigService.mypyWarnUntypedFields == null
        this.warnUntypedFieldsCheckBox.isEnabled = warnUntypedFieldsCheckBox
        ifEnabledRaiseATextPane.isEnabled = warnUntypedFieldsCheckBox
        setHyperlinkHtml(
            textPane1,
            "See <a href=\"https://koxudaxi.github.io/pydantic-pycharm-plugin/\">documentation</a> for more details.</p>"
        )
    }

    val initTyped: Boolean
        get() = initTypedCheckBox.isSelected
    val warnUntypedFields: Boolean
        get() = warnUntypedFieldsCheckBox.isSelected

    private fun setHyperlinkHtml(jTextPane: JTextPane, html: String) {
        jTextPane.contentType = "text/html"
        jTextPane.isEditable = false
        jTextPane.editorKit = HTMLEditorKitBuilder.simple()
        val kit = jTextPane.editorKit
        if (kit is HTMLEditorKit) {
            val css = kit.styleSheet
            css.addRule("a, a:link {color:#" + ColorUtil.toHex(JBUI.CurrentTheme.Link.Foreground.ENABLED) + ";}")
            css.addRule("a:visited {color:#" + ColorUtil.toHex(JBUI.CurrentTheme.Link.Foreground.VISITED) + ";}")
            css.addRule("a:hover {color:#" + ColorUtil.toHex(JBUI.CurrentTheme.Link.Foreground.HOVERED) + ";}")
            css.addRule("a:active {color:#" + ColorUtil.toHex(JBUI.CurrentTheme.Link.Foreground.PRESSED) + ";}")
            css.addRule("body {background-color:#" + ColorUtil.toHex(JBUI.CurrentTheme.DefaultTabs.background()) + ";}")
        }
        jTextPane.text = html
        jTextPane.caret = EmptyCaret.INSTANCE
        jTextPane.background = UIUtil.TRANSPARENT_COLOR
        jTextPane.isOpaque = false
        jTextPane.addHyperlinkListener { e: HyperlinkEvent ->
            if (e.eventType == HyperlinkEvent.EventType.ACTIVATED) {
                BrowserUtil.browse(e.url)
            }
        }
    }
}